src/types.ts

```diff
@@ -9,3 +9,29 @@ export type Options = {
   ASSET_MANIFEST: any
   mapRequestToAsset: (req: Request) => Request
 }
+
+class KVError extends Error {
+  constructor(message?: string, code: number = 500) {
+    super(message)
+    // see: typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html
+    Object.setPrototypeOf(this, new.target.prototype) // restore prototype chain
+    this.name = KVError.name // stack traces display correctly now
+    this.code = code
+  }
+  code: number
+}
+export class MethodNotAllowedError extends KVError {
```

Comment:
i dig these. the code attribute still feels somehow off, and even a bit... vestigial? prescriptive even? if it's going to be an http status code, i would almost prefer we name it that, like `this.httpStatusCode`. `error.code` says something subtly different from a convention standpoint.

I'm not about to die on this hill, btw. the counterpoint to my argument is that having some kind of code will absolutely be helpful to vanilla js users; but if it is intended to represent the status code we think they should return with their response, i think we should call it that.

Reply:
How about let's stick to what Fetch API does for Response and call it `status` ?

Reply:
that sounds like a good compromise 😄